libbpf-rs/src/map.rs

```diff
@@ -375,6 +375,41 @@ impl Map {
         }
     }
 
+    /// Returns whether map is pinned or not flag
+    pub fn is_pinned(&self) -> Result<bool> {
+        match self.ptr {
+            Some(ptr) => Ok(unsafe { libbpf_sys::bpf_map__is_pinned(ptr.as_ptr()) }),
+            None => Err(Error::InvalidInput(("No map pointer found").to_string())),
+        }
+    }
+
+    /// Returns the pin_path if the map is pinned, otherwise, returns an error
+    /// if map is not pinned
+    pub fn get_pin_path(&self) -> Result<&OsStr> {
+        let pinned = match self.is_pinned() {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(Error::InvalidInput(format!(
+                    "get map pin status failed: {e}"
+                )))
+            }
+        };
+
+        if !pinned {
+            return Err(Error::InvalidInput(format!(
+                "map {} is not pinned",
+                self.name()
+            )));
+        }
+
+        let path_ptr = match self.ptr {
+            Some(ptr) => unsafe { libbpf_sys::bpf_map__pin_path(ptr.as_ptr()) },
+            None => return Err(Error::InvalidInput(("No map pointer found").to_string())),
+        };
+        let path_c_str = unsafe { CStr::from_ptr(path_ptr) };
```

Comment:
Can't `bpf_map__pin_path` return `NULL`? It almost feels as if we shouldn't include any pinned checks here. `libbpf` should be the source of truth. So it would probably be better to return an `Option<&OsStr>`, I would say. What do you think?

Reply:
> Can't `bpf_map__pin_path` return `NULL`? It almost feels as if we shouldn't include any pinned checks here. `libbpf` should be the source of truth. So it would probably be better to return an `Option<&OsStr>`, I would say. What do you think?

Yes, it's better to ignore pinned checks.